cdap-client/src/main/java/co/cask/cdap/client/ApplicationClient.java

```diff
@@ -526,4 +568,28 @@ private void deployApp(Id.Namespace namespace, File jarFile, Map<String, String>
 
     return ObjectResponse.fromJsonBody(response, ApplicationDetail.class).getResponseObject().getPrograms();
   }
+
+  /**
+   * Lists programs belonging to an application.
+   *
+   * @param app the application
+   * @return List of all {@link ProgramRecord}s
+   * @throws ApplicationNotFoundException if the application with the given ID was not found
+   * @throws IOException if a network error occurred
+   * @throws UnauthenticatedException if the request is not authorized successfully in the gateway server
+   */
+  public List<ProgramRecord> listPrograms(ApplicationId app)
+    throws ApplicationNotFoundException, IOException, UnauthenticatedException, UnauthorizedException {
+
+    String path = String.format("apps/%s/versions/%s", app.getApplication(), app.getVersion());
```

Comment:
Is it possible to use this API to list the programs of an application deployed via the non-versioned APIs?